### ISSUE TITLE
How to release a Windows-based desktop app from the MacOS-based Android Studio?

### ISSUE THREAD

**Issue by jims57** [flutter/flutter#59735]
![image](https://user-images.githubusercontent.com/1920873/84974712-618a9f80-b156-11ea-9ca6-254401843d7c.png)
![image](https://user-images.githubusercontent.com/1920873/84974889-bc23fb80-b156-11ea-94bc-5916d0139bbc.png)

**Comment by TahaTesser** [comment]
Hi @jims57 
You cannot build windows app on Mac. You need Windows machine for this Please follow [this](https://dev.to/droidmakk/flutter-for-desktop-2on5)

Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by jims57** [comment]
@TahaTesser I don't think it is convenient enough for MacOS users to release Windows-based desktop application, since most of them, including me, don't even use computers or laptops running Windows OS. Even they have, it may waste time installing another Android Studio on another system.

Do you all even consider making MacOS-based Android Studio able to release Windows desktop app directly?

**Comment by TahaTesser** [comment]
@jims57 
I agree, it could be lots of hassle to build windows without a windows machine especially if you're just Mac user
Maybe we can build windows app Visual Studio for Mac (Thanks @VladyslavBondarenko for pointing this)

**Comment by stuartmorgan** [comment]
> I don't think it is convenient enough for MacOS users to release Windows-based desktop application, since most of them, including me, don't even use computers or laptops running Windows OS.

How do you plan to test a Windows-based application before releasing it without the ability to run Windows?

> Even they have, it may waste time installing another Android Studio on another system.

Android Studio isn't used in building a Flutter app for Windows. You do not need to install it to do desktop builds.

> Do you all even consider making MacOS-based Android Studio able to release Windows desktop app directly?

Currently we're using a VS/MSBuild toolchain on Windows, which isn't available for macOS. Supporting cross-compilation would require a completely different approach to building, and add a lot of complexity to the tool. There are other solutions to this problem (VM, which would allowing for actually testing the thing you are planning on releasing), cloud builds (e.g., GitHub  Actions) which already exist and don't require massive changes to the Flutter tooling.

I'll leave this open as a feature request, but we currently have no plans to implement cross-compilation for desktop.

**Comment by stuartmorgan** [comment]
@TahaTesser This isn't related at all to the framework, nor is it a documentation issue. It's a request for a substantial new tool feature.

**Comment by jims57** [comment]
@TahaTesser @stuartmorgan Thanks a lot. I understand the complexity involving building it on MacOS, just ask if you have the plan or not. That is fine with me now.

**Comment by fzyzcjy** [comment]
Same issue here. If using a windows virtual machine on macos to *build* the windows app, it can be slow because of the performance penalty of the VM. Hope this can be implemented!

**Comment by stuartmorgan** [comment]
This is blocked on https://github.com/flutter/flutter/issues/77638, so anyone interested in making this possible would need to start there.
